reporter/src/main/kotlin/reporters/AttributionDocumentPdfReporter.kt

```diff
@@ -0,0 +1,134 @@
+import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.utils.AttributionDocumentPdfModel
+import org.ossreviewtoolkit.spdx.SpdxLicense
+import org.ossreviewtoolkit.utils.ProcessedDeclaredLicense
+import org.ossreviewtoolkit.utils.safeDeleteRecursively
+import org.ossreviewtoolkit.utils.toHexString
+
+private const val DEFAULT_TEMPLATE_ID = "basic-pdf-template"
+private const val TEMPLATE_ID = "template.id"
+private const val TEMPLATE_PATH = "template.path"
+
+class AttributionDocumentPdfReporter : Reporter {
+    override val reporterName = "AttributionDocumentPDF"
+    override val defaultFilename = "attribution-document.pdf"
+
+    override fun generateReport(outputStream: OutputStream, input: ReporterInput, options: Map<String, String>) {
+        val ortResult = input.ortResult
+        val artifacts = ortResult.getPackages().map { (pkg, _) ->
+            AttributionDocumentPdfModel(
+                purl = pkg.purl,
+                binaryFilename = pkg.binaryArtifact.takeUnless { it.url.isEmpty() }?.let { File(it.url).name },
+                declaredLicenses = createLicenseInfo(pkg.declaredLicensesProcessed, input.licenseTextProvider),
+                copyrightStatements = createCopyrightStatement(pkg.id, ortResult, input.packageConfigurationProvider)
+            )
+        }.toList()
+
+        val rootProject = input.ortResult.getProjects().singleOrNull()
+
+        // TODO: Add support for multiple projects.
+        requireNotNull(rootProject) {
+            "The $reporterName currently only supports ORT results with a single project."
+        }
+
+        val projectCopyright = createCopyrightStatement(rootProject.id, ortResult, input.packageConfigurationProvider)
+
+        // Use the default template unless a custom template is provided via the options.
+        var templateId = DEFAULT_TEMPLATE_ID
+        options[TEMPLATE_ID]?.let { id ->
+            options[TEMPLATE_PATH]?.let { path ->
+                templateId = id
+                addTemplateToClasspath(File(path).toURI().toURL())
+            }
+        }
+
+        val workingDir = createTempDir()
+        val documentFile = AttributionDocumentGeneratorImpl(
+            defaultFilename,
+            workingDir,
+            templateId,
+            DocumentValues(rootProject.id.name, rootProject.id.version, projectCopyright)
+        ).generate(artifacts)
+
+        if (documentFile.isFile) {
+            documentFile.inputStream().use {
+                it.copyTo(outputStream)
+            }
+        }
+
+        workingDir.safeDeleteRecursively()
+    }
+
+    private fun createCopyrightStatement(
+        id: Identifier,
+        ortResult: OrtResult,
+        packageConfigurationProvider: PackageConfigurationProvider
+    ) =
+        ortResult.getDetectedLicensesWithCopyrights(id, packageConfigurationProvider)
+            .values
+            .flatten()
+            .joinToString("\n")
```

Comment:
Shouldn't you filter the copyrights of the declared licenses here? It seems inconsistent to add only the declared licenses, but then also add the copyrights for all detected licenses.